integration-tests/src/test/java/eu/xenit/alfresco/healthprocessor/integrationtest/solr/SolrIndexFixerIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+    private static final Duration SOLR_INDEXING_MAX_WAIT = SOLR_INDEXING_WAIT_DEFAULT.multipliedBy(4);
+    private static final Duration SOLR_INDEXING_POLL_INTERVAL = Duration.ofSeconds(1);
+    private static final Duration HEALTH_PROCESSOR_MAX_WAIT = Duration.ofSeconds(40);
+    private static final Duration HEALTH_PROCESSOR_POLL_INTERVAL = Duration.ofSeconds(1).dividedBy(2);
+
+    private static final String SOLR_PLUGIN_NAME = "SolrIndexValidationHealthProcessorPlugin";
+
+    @Test
+    void reindexesNodesAfterPurge() {
+        // Disable health fixer plugin
+        setHealthFixerPlugin(false);
+
+        // Wait until nodes are all indexed in solr
+        waitUntilNodesIndexed("Wait for initial index to be completed", EXPECTED_INDEXED_NODES, SOLR_INDEXING_MAX_WAIT);
+
+        // Wait for the health processor to become IDLE and it has had at least one iteration
+        await("Until some health reports have been received")
+                .atMost(HEALTH_PROCESSOR_MAX_WAIT)
+                .pollInterval(HEALTH_PROCESSOR_POLL_INTERVAL)
+                .until(() -> getHealthProcessorReport(NodeHealthStatus.HEALTHY), greaterThan(0L));
+        waitUntilHealthProcessorIdle("Wait for health processor being finished to record number of healthy nodes");
+        long allHealthyNodes = getHealthProcessorReport(NodeHealthStatus.HEALTHY);
+
+        // Purge nodes from solr index
+        purgeNodes();
+        // Wait until there are no more indexed nodes (they have been purged from the index by solr in the next maintenance interval)
+        waitUntilNodesIndexed("Wait until nodes have been purged from the index", 0L);
+
+        // Wait until our health checker has detected less healthy nodes than before
+        await("Until non-healthy nodes are detected")
+                .atMost(HEALTH_PROCESSOR_MAX_WAIT)
+                .pollInterval(HEALTH_PROCESSOR_POLL_INTERVAL)
+                .until(() -> getHealthProcessorReport(NodeHealthStatus.HEALTHY), lessThan(allHealthyNodes));
+
+        // Then wait until health processor is IDLE
+        waitUntilHealthProcessorIdle("Until health processor cycle has completed to detect number of fixed nodes");
+
+        // Check that unhealthy reports are received for unindexed nodes
+        assertThat(getHealthProcessorReport(NodeHealthStatus.UNHEALTHY), equalTo(EXPECTED_INDEXED_NODES));
+
+        // Enable health fixer plugin again
+        setHealthFixerPlugin(true);
+
+        // Wait until our health checker has re-run and has fixed the issues
+        await("Until fixed nodes are detected")
+                .atMost(HEALTH_PROCESSOR_MAX_WAIT)
+                .pollInterval(HEALTH_PROCESSOR_POLL_INTERVAL)
+                .until(() -> getHealthProcessorReport(NodeHealthStatus.FIXED), equalTo(EXPECTED_INDEXED_NODES));
+
+        // And then that the purged nodes are indexed again
+        waitUntilNodesIndexed("Wait until fixed nodes have been reindexed", EXPECTED_INDEXED_NODES);
```

Comment:
And after that let the healthprocessor run again to see all healthy nodes ? Check if the reindex really works

Reply:
Well, if this request can find the nodes, they have been indexed succesfully